src/Scope.ts

```diff
@@ -503,6 +505,46 @@ export class Scope {
         }
     }
 
+    /**
+    * Find function parameters and function return types that are neither built-in types or known Class references
+    */
+    private diagnosticDetectInvalidFunctionExpressionTypes(file: BscFile) {
+        const classMap = this.getClassMap();
+        for (let func of file.parser.references.functionExpressions) {
+            if (func.returnType instanceof CustomType) {
```

Comment:
Don't use `instanceof` because it won't work for AST provided by a different binary version of brighterscript (i.e. v0.23.0 and v0.23.1 would return false for instanceof). Instead, add a new function in `src/astUtils/reflection.ts` like this: https://github.com/rokucommunity/brighterscript/blob/b69b86259fed32b69841d87c04ca107deb2f3dbc/src/astUtils/reflection.ts#L227-L229